gtsfm/densify/mvs_patchmatchnet.py

```diff
@@ -0,0 +1,280 @@
+            geo_pixel_thresh: geometric pixel threshold
+            geo_depth_thresh: geometric depth threshold
+            photo_thresh: photometric threshold
+
+        Returns:
+            3D coordinates (in the world frame) of the dense point cloud, (point number, 3)
+        """
+        # coordinates of the final point cloud
+        vertices = []
+        # vertex colors of the final point cloud, used in generating colored mesh
+        vertex_colors = []
+
+        packed_pairs = dataset.get_packed_pairs()
+
+        # For each reference view and the corresponding source views
+        for pair in packed_pairs:
+            ref_view = pair["ref_id"]
+            src_views = pair["src_ids"]
+
+            # Load the camera parameters
+            ref_intrinsics, ref_extrinsics = dataset.get_camera_params(ref_view)
+
+            # Load the reference image
+            ref_img = dataset.get_image(ref_view)
+            # Load the estimated depth of the reference view
+            ref_depth_est = depth_list[ref_view][0]
+            # Load the photometric mask of the reference view
+            confidence = confidence_list[ref_view]
+            # Filter the pixels that satisfy photometric consistancy among reference view and source views,
+            #   by checking whether the confidence is larger than the pre-defined photometric threshold
+            photo_mask = confidence > photo_thresh
+
+            all_srcview_depth_ests = []
+
+            # Compute the geometric mask, the value of geo_mask_sum means the number of source views where
+            #   the reference depth is valid according to the geometric thresholds
+            geo_mask_sum = 0
+            for src_view in src_views:
+                # camera parameters of the source view
+                src_intrinsics, src_extrinsics = dataset.get_camera_params(src_view)
+
+                # the estimated depth of the source view
+                src_depth_est = depth_list[src_view][0]
+
+                # Check geometric consistency
+                geo_mask, depth_reprojected, _, _ = patchmatchnet_eval_funcs.check_geometric_consistency(
+                    ref_depth_est,
+                    ref_intrinsics,
+                    ref_extrinsics,
+                    src_depth_est,
+                    src_intrinsics,
+                    src_extrinsics,
+                    geo_pixel_thresh,
+                    geo_depth_thresh,
+                )
+                geo_mask_sum += geo_mask.astype(np.int32)
+                all_srcview_depth_ests.append(depth_reprojected)
+
+            depth_est_averaged = (sum(all_srcview_depth_ests) + ref_depth_est) / (geo_mask_sum + 1)
+            # Valid points requires at least 3 source views validated under geometric threshoulds
+            geo_mask = geo_mask_sum >= DEFAULT_MINIMUM_CONSISTENT_VIEW_NUMBER
+
+            # Combine geometric mask and photometric mask
+            final_mask = np.logical_and(photo_mask, geo_mask)
```

Comment:
i probably prefer the name `joint_mask` instead of `final_mask`, since it's more descriptive

Reply:
resolved, thanks